include/fms_platform.h

```diff
@@ -80,39 +52,28 @@ use,intrinsic :: iso_c_binding, only: c_double,c_float,c_int64_t, &
 !DEC$ MESSAGE:'Using cray pointers.'
 #endif
 
-
-!Control size of integers that will hold address values.
-!Appears for legacy reasons, but seems rather dangerous.
-#ifdef _32bits
-#define POINTER_KIND 4
-!DEC$ MESSAGE:'Using 4-byte addressing'
-#endif
-
-
-!If you do not want to use 64-bit integers.
-#ifdef no_8byte_integers
-#define LONG_KIND INT_KIND
-#endif
-
-
-!If you do not want to use 32-bit floats.
-#ifdef no_4byte_reals
-#define FLOAT_KIND DOUBLE_KIND
-#define NF_GET_VAR_REAL nf_get_var_double
-#define NF_GET_VARA_REAL nf_get_vara_double
-#define NF_GET_ATT_REAL nf_get_att_double
-#undef OVERLOAD_R4
-#undef OVERLOAD_C4
-#endif
-
-
-!If you want to use quad-precision.
-#define NO_QUAD_PRECISION
-#if defined(QUAD_PRECISION) && !defined(__PGI)
-#undef NO_QUAD_PRECISION
+!Set type kinds.
+#ifdef PORTABLE_KINDS
+use,intrinsic :: iso_fortran_env, only: real128
+use,intrinsic :: iso_c_binding, only: c_double,c_float,c_int64_t, &
+                                      c_int32_t,c_int16_t,c_intptr_t
+#define QUAD_KIND real128
+#define DOUBLE_KIND c_double
```

Comment:
@menzel-gfdl is there a reason we use the values in iso_c_binding instead of iso_fortran_env for everything except `QUAD_KIND`?

Reply:
I chose to use the c binding kinds because we do call a little bit of c code from Fortran (i.e. affinity.c).  For quad precision, I could have used "c_long_double", but in c "the only guarantee is that long double is not smaller than double, which is not smaller than float".  This same argument obviously also applies to doubles and floats, so if we care more about having variables be an exact size than being interoperable with c we can change DOUBLE_KIND and FLOAT_KIND to use the Fortran kinds too.

Reply:
Does the fortran code use any of this, or just C?  Should we have different macros for fortran and C?

Reply:
Since we have a `use` clause in the include file, this file is _only_ Fortran.  We need to rename this file to have a different extension as `.h` is specifically a C style header file.  We use `.inc` in other locations, so we should likely switch to that.  (Note: The Fortran standard does not define file extensions.)

Reply:
The Fortran standard does not define extensions, but many compilers and tools do. *.F90 should be used for Fortran code that needs to be subjected to a pre-processor (as is normal for Fortran these days).

Code should not be kept in .inc files. Instead, use *.F90 fils. Once mkmf is not used this will b no problem. Until then, we can hack together something if needed. But the end goal should be to have your Fortran code in *.F90 files, with include-style info in .inc files.

Reply:
If this isn't being used by ant of the C files, then this goes back to my original point of questioning the iso_c_binding values.  Do these interface with C at all?  Should we change them to use the iso_fortran_env values if this is a fortran only file?

Reply:
While this file is not used directly by any C/C++ code, it is necessary to define Fortran datatypes for consistency when interfacing Fortran and C/C++.